device/iot-device-samples/send-serialized-event/src/main/java/samples/com/microsoft/azure/sdk/iot/SendSerializedEvent.java

```diff
@@ -82,47 +112,46 @@ public void execute(IotHubStatusCode status, Object context)
         {
             System.out.println("IoT Hub responded to message with status " + status.name());
 
-            if (context != null) {
-                synchronized (context) {
-                    context.notify();
```

Comment:
Wait and notify pattern doesn't fit well with the context style callback approach our library has. Countdownlatch is a bit easier to use here